### ISSUE TITLE
Astro actions cause warning in dev mode for static pages

### ISSUE THREAD

**Issue by hkbertoson** [withastro/astro#11751]
### Astro Info

```block
Astro                    v4.14.2
Node                     v18.20.3
System                   Linux (x64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/node
```

### Describe the Bug

Same issue that was reported here #11099. Looks like it was fixed in #11111 

But its back yet again. 

### What's the expected result?

No warning should show for prerendered pages

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-6bjgpz

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bholmesdev** [comment]
Hey @hkbertoson! Thanks for reporting. Pesky warnings strike again. I'll see where we're tripping that wire.

**Comment by hfournier** [comment]
I get this warning:
```
[WARN] `Astro.request.headers` is unavailable in "static" output mode, and in prerendered pages within "hybrid" and "server" output modes. If you need access to request headers, make sure that `output` is configured as either `"server"` or `output: "hybrid"` in your config file, and that the page accessing the headers is rendered on-demand.
```
by simply adding:
```
experimental: {
		actions: true
	},
```
Everything seems to work as expected.

I've tried commenting out the actions related code and the warning remains until I comment out the experimental flag.

**Comment by hkbertoson** [comment]
The PR has been merged. Just not released yet. Should be gone once it's been released.

**Comment by eric-burel** [comment]
I got tricked by this warning when making another mistake, I forgot to use "client:load" directive on the client component to enable the "onSubmit" callback and prevent the default form behaviour from reloading the page.

**Comment by bholmesdev** [comment]
Just merged the release PR! Expect this to land in v4.14.4
